cirq-core/cirq/devices/noise_properties.py

```diff
@@ -234,37 +42,69 @@ def __init__(self, noise_properties: NoiseProperties) -> None:
         Raises:
             ValueError: if no NoiseProperties object is specified.
         """
-        if noise_properties is not None:
-            self._noise_properties = noise_properties
-        else:
-            raise ValueError('A NoiseProperties object must be specified')
+        self._noise_properties = noise_properties
+        self.noise_models = self._noise_properties.build_noise_models()
 
-    def noisy_moment(
-        self, moment: ops.Moment, system_qubits: Sequence['cirq.Qid']
-    ) -> 'cirq.OP_TREE':
-        moments: List[ops.Moment] = []
+    def virtual_predicate(self, op: 'cirq.Operation') -> bool:
+        """Returns True if an operation is virtual.
 
-        if any(
-            [protocols.is_measurement(op.gate) for op in moment.operations]
-        ):  # Add readout error before measurement gate
-            p00 = self._noise_properties.p00
-            p11 = self._noise_properties.p11
-            measurement_qubits = [
-                list(op.qubits)[0] for op in moment.operations if protocols.is_measurement(op.gate)
+        Device-specific subclasses should implement this method to mark any
+        operations which their device handles outside the quantum hardware.
+        """
+        return False
```

Comment:
Should this be False always ?

Reply:
In the general case, yes - outside of the Google device (which gets its own implementation of this), what gates are "virtual" varies between devices. I suppose we could check for `VirtualTag` here, but note that we don't use `VirtualTag` in the Google version (see google_noise_properties in #4733).

Reply:
Discussed offline: keep the current behavior, but document it thoroughly (especially the virtual/non-virtual distinction).